src/main/java/com/flowingcode/vaadin/addons/datetimerangepicker/api/DateTimeRange.java

```diff
@@ -0,0 +1,373 @@
+/*-
+ * #%L
+ * DateTimeRangePicker Add-on
+ * %%
+ * Copyright (C) 2025 Flowing Code
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.flowingcode.vaadin.addons.datetimerangepicker.api;
+
+import java.io.Serializable;
+import java.time.DayOfWeek;
+import java.time.Duration;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.Period;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * A class to operate {@link TimeInterval} instances based on defined date and time constraints
+ *
+ * @author Izaguirre, Ezequiel
+ * @see TimeInterval
+ */
+public class DateTimeRange implements Serializable {
+
+  private static final DayOfWeek[] defaultWeekDays = DayOfWeek.values();
+  private static final LocalTime defaultStartTime = LocalTime.MIN;
+  private static final LocalTime defaultEndTime = LocalTime.MAX;
+
+  private final LocalDate startDate;
+  private final LocalDate endDate;
+  private final TreeSet<DayOfWeek> weekDays = new TreeSet<>();
+  private LocalTime startTime = defaultStartTime;
+  private LocalTime endTime = defaultEndTime;
+
+
+  public DateTimeRange(LocalDate startDate, LocalDate endDate, Set<DayOfWeek> weekDays) {
+    if (!startDate.isBefore(endDate)) {
+      throw new IllegalArgumentException("startDate must be before endDate");
+    }
+    this.startDate = startDate;
+    this.endDate = endDate;
+    setWeekDays(weekDays);
+  }
+
+  public DateTimeRange(LocalDate startDate, LocalDate endDate) {
+    this(startDate, endDate, Set.of(defaultWeekDays));
+  }
+
+  public DateTimeRange(LocalDate startDate, LocalDate endDate, LocalTime startTime, LocalTime endTime,
+      Set<DayOfWeek> weekDays) {
+    this(startDate, endDate, weekDays);
+    setDayDuration(startTime, endTime);
+  }
+
+  public DateTimeRange(LocalDate startDate, LocalDate endDate, LocalTime startTime, LocalTime endTime) {
+    this(startDate, endDate, startTime, endTime, Set.of(defaultWeekDays));
+  }
+
+  /**
+   * Sets time boundaries for the intervals
+   *
+   * @param startTime the starting point
+   * @param endTime   the ending point (exclusive).
+   */
```

Comment:
Just one more comment before moving forward with this. There are several places where the "main descripcion" part of the Javadocs is missing the closing period and other places where the "param description" in ending with one when it shouldn't. Please see https://github.com/FlowingCode/DevelopmentConventions/blob/main/code-style.md#javadoc and make the necessary adjustments here and review the rest.